api_client/python/timesketch_api_client/client.py

```diff
@@ -365,102 +367,232 @@ def _create_session(
         return session
 
     def fetch_resource_data(self, resource_uri, params=None):
-        """Make a HTTP GET request.
+        """Makes an HTTP GET request to the specified resource URI with retries.
+
+        This method attempts to fetch data from the Timesketch API. It implements
+        a manual retry mechanism with a fixed 1-second backoff between attempts
+        if the initial request fails or returns an empty (but valid JSON) response.
+        Retries occur for network connection errors, API errors (non-20x status codes),
+        JSON decoding errors, or if the API returns a successful (20x) response
+        with a "falsy" JSON payload (e.g., null, empty list/dictionary).
+
 
         Args:
             resource_uri (str): The URI to the resource to be fetched.
-            params (dict): Dict of URL parameters to send in the GET request.
+            params (dict, optional): A dictionary of URL parameters to send
+                in the GET request. Defaults to None.
 
         Returns:
-            Dictionary with the response data.
+            dict: A dictionary containing the JSON response data from the API.
 
         Raises:
-            ValueError: If response could not be JSON-decoded after
-                DEFAULT_RETRY_COUNT attempts.
-            RuntimeError: If the API server returns an error or empty.
+            requests.exceptions.ConnectionError: If a connection error persists
+                after all retry attempts.
+            ValueError: If the API response cannot be JSON-decoded after all
+                retry attempts.
+            RuntimeError: If the API server returns an error (non-20x status code)
+                or a "falsy" JSON response (e.g., null, empty list/dict)
+                after all retry attempts.
         """
         resource_url = "{0:s}/{1:s}".format(self.api_root, resource_uri)
 
-        retry_count = 0
+        # Start attempt count from 0, first loop with set it to 1
+        attempt = 0
         result = None
         while True:
-            retry_count += 1
-            response = self.session.get(resource_url, params=params)
+            attempt += 1
+            # If this is not the first attempt, wait before trying again
+            if attempt > 1:
+                # Simple fixed backoff before retrying
+                time.sleep(1)  # Wait for 1 second before the next attempt
```

Comment:
happy to also make the exponential sleep here, but I was not sure if that is needed.

Reply:
Let's do an exponential backoff here if we already touch this code.